pyfftw/interfaces/dask_fft.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python
+#
+# Henry Gomersall
+# heng@kedevelopments.co.uk
+#
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+# list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+# this list of conditions and the following disclaimer in the documentation
+# and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its contributors
+# may be used to endorse or promote products derived from this software without
+# specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+
+'''
+This module implements those functions that replace aspects of the
+:mod:`dask.fft` module. This module *provides* the entire documented
+namespace of :mod:`dask.fft`, but those functions that are not included
+here are imported directly from :mod:`dask.fft`.
+
+
+It is notable that unlike :mod:`numpy.fftpack`, which :mod:`dask.fft`
+wraps, these functions will generally return an output array with the
+same precision as the input array, and the transform that is chosen is
+chosen based on the precision of the input array. That is, if the input
+array is 32-bit floating point, then the transform will be 32-bit floating
+point and so will the returned array. Half precision input will be
+converted to single precision.  Otherwise, if any type conversion is
+required, the default will be double precision.
+
+The exceptions raised by each of these functions are mostly as per their
+equivalents in :mod:`dask.fft`, though there are some corner cases in
+which this may not be true.
+'''
+
+from . import numpy_fft
```

Comment:
It's probably cleaner to import functions from this interface, wrap them, and overwrite the associate variables with our wrappers. This was we don't accidentally expose this module in our interface. PR ( https://github.com/pyFFTW/pyFFTW/pull/165 ) does exactly this.